src/borg/repository.py

```diff
@@ -1526,7 +1526,15 @@ def _read(self, fd, fmt, header, segment, offset, acceptable_tags, read_data=Tru
                 size, segment, offset))
         length = size - fmt.size
         if read_data:
-            data = fd.read(length)
+            try:
+                data = fd.read(length)
+            except OSError as err:
+                if err.errno == errno.EIO:
+                    raise IntegrityError(
+                        'I/O error while reading segment file [segment {}, offset {}] : {}'
+                        .format(segment, offset, fd.name)
+                    )
+                raise err
```

Comment:
just "raise"?

Reply:
i assumed that you'd prefer the explicit variant and wasn't sure whether Python 2.7 already knows this syntax.

Reply:
borg is py3 only.